### ISSUE TITLE
"Couldn't fetch" error on Google Search Console

### ISSUE THREAD

**Issue by emretepedev** [nuxt-community/sitemap-module#240]
### Description

I am able to add @nuxtjs/sitemap with nuxt and works like a charm.
![image](https://user-images.githubusercontent.com/70508617/153787191-81af0f2a-ae93-4d92-bc65-70835ed5c7c4.png)

Now I am trying to add sitemap to the Google Search Console. But it failed with an error "Couldn't fetch".
![image](https://user-images.githubusercontent.com/70508617/153787407-cb5c32e8-b699-478a-bb82-c90fca278107.png)

If I click details, Google says "Sitemap could not be read."
![image](https://user-images.githubusercontent.com/70508617/153787644-a81deb1a-6654-439f-bca1-60438e3fcfa1.png)

### Expected behavior

Since @nuxtjs/sitemap has been added to my website, associating sitemap on Google Search Console also must work.

### Actual behavior

I got the errors on Google Search Console. ("Sitemap could not be read" or "Couldn't fetch")

### Environment

@nuxtjs/sitemap@2.4.0
@nuxtjs/robots@2.5.0
nuxt@2.15.8

### Extra

My installation snippet in "nuxt.config.js"

```js
{
// ...
  modules: [
    '@nuxtjs/sitemap',
    '@nuxtjs/robots',
  ],

  sitemap: {
    hostname: process.env.PROJECT_URL,
    gzip: true,
    trailingSlash: false,
    cacheTime: 1000 * 60 * 15,
    defaults: {
      changefreq: 'monthly',
      priority: 0.5,
      lastmod: new Date(),
    }
  },

  robots: [
    {
      UserAgent: '*',
      Allow: '/',
    },
    {
      Sitemap: process.env.PROJECT_URL + '/sitemap.xml',
    },
  ],
// ...
}
```

And my robots.txt file contains that:

```
User-agent: *
Allow: /
Sitemap: <MY_PROJECT_URL>/sitemap.xml
```

**Comment by florian-lefebvre** [comment]
I'm encountering the same issue, my sitemap hadn't been fetched since February 16th. Using Nuxt 2 universal static

**Comment by victorcngo** [comment]
same issue here! just can't fetch my sitemap at all (using Nuxt 2 in SSG just like @florian-lefebvre). only difference is that i created my sitemap with a sitemap generator and put it in /static folder.

**Comment by jnawaz** [comment]
> I'm also experiencing issues with the sitemap module in relation to a production site. 
> 
> When I run locally I can view the sitemap.xml, however publishing to AWS amplify and trying to access the production /sitemap.xml I get redirected to the homepage. 
> 
> I'm still investigating if there is anything I can do to prevent the redirect to the homepage. However seeing the same issues with Google search console. 
> 
> Happy to provide further info if it helps
> 

Update: I managed to get it working in production. I had to amend the redirects and rewrites to allow xml. Once I had updated this, the sitemap worked and Google search console shows it as successfully processed.

**Comment by emretepedev** [comment]
any updates? btw I deployed my site on GitHub pages.

**Comment by Matb85** [comment]
+1 can't fetch my sitemap

**Comment by zoltalar** [comment]
I am also experiencing this issue. Any update?

**Comment by dadasmash** [comment]
same here........

**Comment by simonmaass** [comment]
we are having the same problem!

**Comment by lArtiquel** [comment]
Still Google says that he can't fetch sitemap... Even thru it's accessible in the browser. Can't figure out why. Tried to validate sitemap using tools - it's valid XML. 
![image](https://user-images.githubusercontent.com/36863599/215326198-5cc31abc-c10d-4826-a065-fa8b42abbabb.png)

**Comment by simonmaass** [comment]
still got the same problem aswell...

**Comment by tha80** [comment]
Me too. And no clue why... Help would be very appreciated.

**Comment by lArtiquel** [comment]
I've posted a question on [StackOverflow](https://stackoverflow.com/questions/75309341/why-does-google-search-couldnt-fetch-a-sitemap?noredirect=1#comment132886724_75309341) and described this issue more in details.

Btw, does s-one have already checked servers access logs? Does GoogleBot really hits the website endpoint when it says that it cannot fetch the sitemap?

**Comment by tha80** [comment]
@lArtiquel: Yes, I checked. Google does actually hit my server directly after submitting the sitemap.xml url. And the response from the server also looked fine (status code 200, valid headers, correct size, etc.).

So from my point of view "Can't fetch" is definitely wrong. Maybe it's that Google for some other unknown reasons doesn't like the sitemap or the response from the server. Or it's just a bug in the search console itself. At least a lot of people are reporting the same thing in various forums and groups (with various other frameworks and servers).

**Comment by szadave** [comment]
+1 for this in 2024. :D

**Comment by nblkmal** [comment]
> > I'm also experiencing issues with the sitemap module in relation to a production site.
> > When I run locally I can view the sitemap.xml, however publishing to AWS amplify and trying to access the production /sitemap.xml I get redirected to the homepage.
> > I'm still investigating if there is anything I can do to prevent the redirect to the homepage. However seeing the same issues with Google search console.
> > Happy to provide further info if it helps
> 
> Update: I managed to get it working in production. I had to amend the redirects and rewrites to allow xml. Once I had updated this, the sitemap worked and Google search console shows it as successfully processed.

Hi can you show example on your workaround?

**Comment by ebanDev** [comment]
Hi ! I have the same issue :/
